R/read_favenv.R

```diff
@@ -0,0 +1,186 @@
+#' et al. (2020) and how
+#'
+#' @export
+#'
+#' @references
+#' - Van Calster H., Cools N., De Keersmaeker L., Denys L., Herr C., Leyssen A.,
+#' Provoost S., Vanderhaeghe F., Vandevoorde B., Wouters J. en M. Raman (2020).
+#' Gunstige abiotische bereiken voor vegetatietypes in Vlaanderen. Rapporten van
+#' het Instituut voor Natuur- en Bosonderzoek 2020 (44). Instituut voor Natuur-
+#' en Bosonderzoek, Brussel. DOI: doi.org/10.21436/inbor.19362510D
+#' - Oosterlynck P., De Saeger S., Leyssen A., Provoost S., Thomaes A.,
+#' Vandevoorde B., Wouters J., & Paelinckx D. (2020). Criteria voor de
+#' beoordeling van de lokale staat van instandhouding van de Natura2000
+#' habitattypen in Vlaanderen. Rapporten van het Instituut voor Natuur- en
+#' Bosonderzoek 2020 (27). Instituut voor Natuur- en Bosonderzoek, Brussel.
+#' DOI: doi.org/10.21436/inbor.14061248
+#'
+#' @examples
+#' \dontrun{
+#' favenv <- read_favenv()
+#' }
+#'
+#' @importFrom assertthat
+#' assert_that
+#' is.string
+#' @importFrom dplyr
+#' as_tibble
+#' filter
+#' rename
+#' if_all
+#' everything
+#' @importFrom fs
+#' is_file
+#' file_exists
+#' dir_create
+#' @importFrom n2khabmon
+#' read_schemes
+read_favenv <- function(
+    file = file.path(
+      locate_n2khab_data(),
+      "10_raw/favenv",
+      "VanCalster_etal_2020_Gunstig_abiotisch_bereik_per_milieuvariabele_en_habitatsubtype_v01_00.txt"
+    ),
+    version = c("01.00"),
+    lang = c("nl")) {
+
+  version <- match.arg(version)
+  lang <- match.arg(lang)
+
+  # if file does not exist, download it from Zenodo
+  if (!fs::file_exists(file)) {
+    assert_that(
+      curl::has_internet(),
+      msg =
+        paste0(
+          "The file does not exists and there is no internet connection.\n",
+          "Aborting attempt to dowload the necessary file from Zenodo."
+        )
+    )
+    doi <- switch(
+      version,
+      `01.00` = "10.5281/zenodo.10533792",
+      "10.5281/zenodo.10533791" # default resolves to latest
+    )
+    dir <- fs::path_dir(file)
+    fs::dir_create(dir)
+    download_zenodo(doi = doi, path = dir, quiet = TRUE)
+  }
+
+  favenv <- read.delim(
+    file = file,
+    header = TRUE,
+    sep = "\t",
+    dec = ",",
+    na.strings = c("NA", ""),
+    blank.lines.skip = TRUE
+  )
+
+  favenv <- favenv |>
+    convertdf_enc(from = "latin1", to = "UTF-8") |>
+    as_tibble() |>
+    rename(
+      type = Habitatsubtype,
+      subtype = Subtype,
+      milieucompartiment = Milieucompartiment,
+      variabele = Variabele,
+      afkorting = Afkorting,
+      eenheid = Eenheid,
+      toetswijze_bepaling = `Toetswijze...bepaling`,
+      teken = Teken,
+      abiotisch_bereik = Abiotisch.bereik,
+      waarde_1 = Waarde.1,
+      waarde_2 = Waarde.2,
+      waarde_num_1 = WaardeNum1,
+      waarde_num_2 = WaardeNum2,
+      n_gunstig = N.gunstig,
+      status = Status,
+      referentie = Referentie,
+      opmerking = Opmerking,
+      wijziging = Wijziging
```

Comment:
All column names of tibbles returned by {n2khab} are in English, regardless of the chosen language for the tibble contents. So best update to English.